src/sentry/integrations/vsts/integration.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import absolute_import
+from sentry.integrations import Integration, IntegrationMetadata
+from sentry.utils.pipeline import NestedPipelineView
+from sentry.identity.pipeline import IdentityProviderPipeline
+from sentry.utils.http import absolute_uri
+DESCRIPTION = """
+VSTS
+"""
+
+metadata = IntegrationMetadata(
+    description=DESCRIPTION.strip(),
+    author='The Sentry Team',
+    issue_url='https://github.com/getsentry/sentry/issues/new?title=VSTS%20Integration:%20&labels=Component%3A%20Integrations',
+    source_url='https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/vsts',
+    aspects={},
+)
+
+
+class VSTSIntegration(Integration):
+    key = 'vsts'
+    name = 'VSTS'
+    metadata = metadata
+
+    setup_dialog_config = {
+        'width': 600,
+        'height': 800,
+    }
+
+    identity_oauth_scopes = frozenset([
+        'vso.code_full',
+        'vso.identity_manage',
+        'vso.work_full',
+    ])
+
+    def get_pipeline_views(self):
+        identity_pipeline_config = {
+            'oauth_scopes': self.identity_oauth_scopes,
+            'redirect_url': absolute_uri('/extensions/vsts/setup/'),
+        }
+
+        identity_pipeline_view = NestedPipelineView(
+            bind_key='identity',
+            provider_key='vsts',
+            pipeline_cls=IdentityProviderPipeline,
+            config=identity_pipeline_config,
+        )
+
+        return [identity_pipeline_view]
+
+    def get_config(self):
+        return [{
+            'name': 'name',
+            'label': 'Name',
+            'type': 'text',
+            'required': True,
+        }]
+
+    def build_integration(self, state):
+        data = state['identity']['data']
+
+        scopes = sorted(self.identity_oauth_scopes)
+
+        return {
+            'name': 'VSTS',
+            'external_id': 'vsts',
```

Comment:
@macqueen `'name'` and `'external_id'` are placeholders to be team/project names that are personalized for the user. there is a concept of a default collection and you probably need to say something like this: `curl -u {username}[:{personalaccesstoken}] https://{account}.VisualStudio.com/DefaultCollection/_apis/projects?api-version=2.0` this https://docs.microsoft.com/en-us/rest/api/vsts/?view=vsts and https://docs.microsoft.com/en-us/rest/api/vsts/core/projects/list should be helpful.